test/functional/scripting.ts

```diff
@@ -125,96 +127,92 @@ describe("scripting", function () {
-        redis.monitor(function (err, monitor) {
-          monitor.on("monitor", function (_, command) {
-            const name = expectedComands.shift();
-            expect(name).to.eql(command[0]);
-            if (!expectedComands.length) {
-              monitor.disconnect();
-              redis.disconnect();
-              done();
-            }
-          });
-          redis.test("bar", function () {
-            redis.test("foo");
-          });
-        });
-      });
+    await redis.test("preload");
+    await redis.script("flush");
+
+    const commands = await getCommandsFromMonitor(redis, 5, async () => {
+      await redis.test("foo");
+      await redis.test("bar");
     });
+
+    const expectedComands = ["evalsha", "eval", "get", "evalsha", "get"];
+    expect(commands.map((c) => c[0])).to.eql(expectedComands);
   });
 
-  it("should load scripts first before execute pipeline", function (done) {
+  it("should load scripts first before execution of pipeline", async function () {
     const redis = new Redis();
 
     redis.defineCommand("testGet", {
       numberOfKeys: 1,
       lua: 'return redis.call("get", KEYS[1])',
     });
 
-    redis.testGet("init", function () {
-      redis.defineCommand("testSet", {
-        numberOfKeys: 1,
-        lua: 'return redis.call("set", KEYS[1], "bar")',
-      });
-      const expectedComands = [
-        "script",
-        "script",
-        "evalsha",
-        "get",
-        "evalsha",
-        "set",
-        "get",
-      ];
-      redis.monitor(function (err, monitor) {
-        monitor.on("monitor", function (_, command) {
-          const name = expectedComands.shift();
-          expect(name).to.eql(command[0]);
-          if (!expectedComands.length) {
-            monitor.disconnect();
-            redis.disconnect();
-            done();
-          }
-        });
-        const pipe = redis.pipeline();
-        pipe.testGet("foo").testSet("foo").get("foo").exec();
-      });
+    await redis.testGet("init");
+
+    redis.defineCommand("testSet", {
+      numberOfKeys: 1,
+      lua: 'return redis.call("set", KEYS[1], "bar")',
+    });
+
+    const commands = await getCommandsFromMonitor(redis, 5, () => {
+      return redis.pipeline().testGet("foo").testSet("foo").get("foo").exec();
+    });
+
+    const expectedComands = ["evalsha", "get", "eval", "set", "get"];
+
+    expect(commands.map((c) => c[0])).to.eql(expectedComands);
+  });
+
+  it("does not fallback to EVAL in transactions", async function () {
+    const redis = new Redis();
+
+    redis.defineCommand("test", {
+      numberOfKeys: 1,
+      lua: 'return redis.call("get", KEYS[1])',
+    });
+
+    await redis.test("preload");
+    await redis.script("flush");
+    const spy = sinon.spy(redis, "sendCommand");
+    const commands = await getCommandsFromMonitor(redis, 4, async () => {
+      const [a, b] = await redis.multi().test("foo").test("bar").exec();
```

Comment:
A test with manual transactions would be nice. 
But I don't think this is supported well at the moment.
e.g.
```js
redis.pipeline([['multi'], ['test', 'foo'], ['exec']]).exec()
```